### DISCUSSION TITLE
CreateWebHashHistory is not able to refresh and redirect the page when editing the path url from the browser

### DISCUSSION THREAD

**Discussion by Meng991003** [vuejs/router#2255] [Help and Questions] [answered]
I have created my router with the below configuration.
```javascript
import { createRouter, createWebHashHistory } from 'vue-router'
import SignInRoutes from 'eb-app-signin/src/signin.routes'

const routes = [
	{
		path: '/',
		redirect: '/mo/eb-app-signin/landing'
	}
].concat(SignInRoutes)

const router = createRouter({
	history: createWebHashHistory(),
	routes
})
```
It working fine when i using createWebHistory, but when I change it to use createWebHashHistory, the browser is not able to redirect and refresh properly.

For example, when i edit the path url to '/mo/eb-app-signin' should redirect user to '/mo/eb-app-signin/landing' which is the expected behaviour. However, when i using createWebHashHistory, the redirection is not able to happen.

**Comment by posva** [comment, marked as answer]
You need to configure your server: https://router.vuejs.org/guide/essentials/history-mode.html#Example-Server-Configurations
